### PR TITLE
[REF] point_of_sale: unify order components

### PR DESCRIPTION
**Pull request by vlst-odoo**
At the moment, the pos displays the selected order in multiple
screens, such as the `ProductScreen`, the `TicketScreen`, the
`SplitBillScreen`, the `self_order` app, but uses completely
different implementations for each. This means that when
a change needs to be made to the one, it generally needs to be made
to the other as well, which is error-prone.

In this commit we create generic `OrderWidget` and `Orderline` components,
that have not dependencies on POS objects and we refactor the `ProductScreen`,
the `TicketScreen` and the `SplitBillScreen` to use these components.

The `OrderSummary` component was removed and it's contents are now in the
`OrderWidget`.

We also introduce a very simple new component called `CenteredIcon` that
can be used whenever we want to show an icon that fills a whole div. In this
commit we use it for showing the cart icon when the order is empty.

In a future commit, the `pos_self_order` will be refactored in order
to use the same `OrderWidget` and `Orderline` compoenents as the POS.

Task: 3470365

In addition, this commit solves the issue from the task 3491303.

https://github.com/odoo/enterprise/pull/45972

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/app/screens/product_screen/product_screen.xml [hunk 1]
```diff
@@ -3,10 +3,34 @@
...
+                        class="{ ...line.getDisplayClasses(), 'selected' : line.selected }">
+                        <t t-set-slot="product-name">
+                            <i  t-if="line.get_product().isTracked()"
+                                t-on-click.stop="() => line.editPackLotLines()" role="img"
+                                t-attf-class="{{ line.has_valid_product_lot() ? 'text-success' : 'text-danger'}} fa fa-list line-lot-icon ms-1"
+                                t-attf-title="{{ line.has_valid_product_lot() ? 'V' : 'Inv'}}alid product lot" />
+                        </t>
+                        <li t-if="line.get_discount_str() !== '0'">
+                            With a <em><t t-esc="line.get_discount_str()" />% </em> discount
+                        </li>
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/screens/product_screen/product_screen.xml:25]: move this to the main orderline comp, above the customer note

File: addons/point_of_sale/static/src/app/generic_components/orderline/orderline.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/** @odoo-module */
+
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/generic_components/orderline/orderline.js:2]: add comment explaining how to extend this comp

File: addons/point_of_sale/static/src/app/screens/product_screen/product_screen.xml [hunk 3]
```diff
@@ -3,10 +3,31 @@
...
+            <div t-if="!ui.isSmall || pos.mobile_pane === 'left'"
+                t-att-class="{'flex-grow-1': ui.isSmall}"
+                class="leftpane d-flex flex-column border-end bg-200" >
+                <OrderWidget lines="currentOrder.orderlines" t-slot-scope="orderline"
+                    total="env.utils.formatCurrency(currentOrder.get_total_with_tax())"
+                    tax="env.utils.roundCurrency(currentOrder.get_total_tax()) and env.utils.formatCurrency(currentOrder.get_total_tax()) or ''">
+                    <t t-set="line" t-value="orderline.line" />
+                    <Orderline line="line.getDisplayData()"
+                        t-on-click="() => this.selectLine(line)"
+                        t-on-dblclick="()=>line.toggleSkipChange()"
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/screens/product_screen/product_screen.xml:14]: move this to restaurant

File: addons/point_of_sale/static/src/app/store/models.js [hunk 4]
```diff
@@ -470,6 +449,12 @@ export class Orderline extends PosModel {
         orderline.customerNote = this.customerNote;
         return orderline;
     }
+    getDisplayClasses() {
+        return {
+            "has-change text-success border-start border-success border-4": this.hasChange,
+            "skip-change text-primary border-start border-primary border-4": this.skipChange,
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/store/models.js:455]: leave this as an empty function and override in resto

File: addons/point_of_sale/static/src/app/generic_components/orderline/orderline.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                <li t-if="line.comboParent" class="combo-parent-name">
+                    <i class="fa fa-th-large me-2" role="img" aria-label="Combo" title="Combo"/>
+                    <t t-esc="line.comboParent" />
+                </li>
+                <t t-slot="default" />
+                <li t-if="line.customerNote" class="customer-note w-50 p-2 my-1 rounded text-bg-warning text-warning bg-opacity-25 text-break">
+                    <i class="fa fa-sticky-note me-1" role="img" aria-label="Customer Note" title="Customer Note"/>
+                    <t t-esc="line.customerNote" />
+                </li>
+                <li t-if="line.internalNote" class="internal-note w-50 p-2 my-1 rounded text-bg-info text-info bg-opacity-25 text-break">
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/generic_components/orderline/orderline.xml:31]: this should be in restaurant

File: addons/l10n_fr_pos_cert/static/src/xml/Orderline.xml [hunk 6]
```diff
@@ -1,14 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates id="template" xml:space="preserve">
```
Comment (vlst-odoo) [inline addons/l10n_fr_pos_cert/static/src/xml/Orderline.xml:2]: change the name of the file

File: addons/pos_loyalty/static/src/overrides/components/order_summary/order_summary.xml [hunk 7]
```diff
@@ -1,36 +1,25 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (vlst-odoo) [inline addons/pos_loyalty/static/src/overrides/components/order_summary/order_summary.xml:1]: change file name

File: addons/pos_sale/static/src/overrides/components/product_screen/orderline/orderline.xml [hunk 8]
```diff
@@ -1,24 +1,21 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (vlst-odoo) [inline addons/pos_sale/static/src/overrides/components/product_screen/orderline/orderline.xml:1]: rename file

File: addons/pos_restaurant/static/src/app/product_screen/order/order.xml [hunk 9]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/app/product_screen/order/order.xml:1]: rename file

**Review by caburj — APPROVED** [review]
robodoo r+

**Pull request closed without merging**